daemon/src/image_picker.rs

```diff
@@ -12,8 +13,9 @@ use crate::{
     wallpaper_info::{Sorting, WallpaperInfo},
 };
 
+#[derive(Debug)]
 struct Queue {
-    buffer: Vec<PathBuf>,
+    buffer: VecDeque<PathBuf>,
     current: usize,
     tail: usize,
```

Comment:
I thought about it but just seemed more tedious to have `queue.size - 1` everywhere. Could just replace it with that if you want though yeah.

> Most `VecDeque` returns a `Option<T>`, maybe we can use that.

Not sure what you mean sorry. As in use `queue.buffer.get(next_index)` instead of checking the index against `tail`?